### PR TITLE
fixes #8781 by appending "_U" instead of 'U'

### LINKED ISSUE #8781: Duplicate member error for union types
I was doing some tests and I noticed that this code here doesn't compile if I add the animatedU field
```nim
# Compile command: nim c -r src/Vamp.nim
Sprite* = ref object of Drawable
    animatedU: bool
    case animated: bool
    of true:
        frames: seq[Frame]
    of false:
        region: Region
```
It gives me the error:
```bash
/src/nimcache/Vamp_items.c:144:3: error: duplicate member ‘animatedU’
 } animatedU;
   ^~~~~~~~~
```
I decided to take a look at the generated C code and I found the issue, here is the generated C code for the object:
```c
struct tyObject_SpritecolonObjectType__eTjdHCb9c3Bxp5ROoIXYUog {
  tyObject_DrawablecolonObjectType__3rGMvEQlWCYWoG9arkDV5cQ Sup;
NIM_BOOL animatedU;
NIM_BOOL animated;
union{
struct {tySequence_ajsOgsn5QRxRS9aOU9a1yVEg* frames;
} S1;
struct {tyObject_RegioncolonObjectType__OOyhagHo0p4ElifghzFabw* region;
} S2;
} animatedU;     // <-- This is the duplicate member
};
```
Not entirely sure if this is a bug, or if its just assumed that the programmer should not have variable names that end in a capital U.

### PR DESCRIPTION
**Pull request by Vindaar**
Fixes #8781 by appending 'U' with a prefixed underscore to avoid clashing with user defined names ending on 'U'.

### CODE REVIEW

**Comment by Vindaar** [comment]
Ok, now I'm a little confused. With this fix your proposed test case still fails for me with the same error.

I thought the idea was that the underscores were removed from variable names, hence it could not result in the same error.
Or in other words, according to the [spec](https://nim-lang.org/docs/manual.html#lexical-analysis-identifiers-keywords) an identifier such as `animated_U` would also be allowed. 

Regardless if anything it should result in a Nim compiler error.

**Comment by LemonBoy** [comment]
What about using `_U` as a prefix since field names cannot start with a underscore?

**Comment by Vindaar** [comment]
Sounds like a good solution to me. @Araq?

File: compiler/ccgtypes.nim [hunk 1]
```diff
@@ -449,7 +449,8 @@ proc genRecordFieldsAux(m: BModule, n: PNode,
   of nkRecCase:
     if n.sons[0].kind != nkSym: internalError(m.config, n.info, "genRecordFieldsAux")
     add(result, genRecordFieldsAux(m, n.sons[0], accessExpr, rectype, check))
-    let uname = rope(mangle(n.sons[0].sym.name.s) & 'U')
+    let uname = rope(mangle(n.sons[0].sym.name.s) & "_U")
```
Comment (LemonBoy) [inline compiler/ccgtypes.nim]: This line should go away (and a few words explaining why such a prefix has been chosen helps here)

File: tests/types/t8781.nim [hunk 2]
```diff
@@ -0,0 +1,25 @@
+discard """
```
Comment (LemonBoy) [inline tests/types/t8781.nim]: This is a `ccgbugs` test, not a `types` one.

File: compiler/ccgtypes.nim (hunk 1 above)
Comment (Vindaar) [reply compiler/ccgtypes.nim]: Oh, I guess I screwed up picking lines to commit, haha.

File: tests/types/t8781.nim (hunk 2 above)
Comment (Vindaar) [reply tests/types/t8781.nim]: Thanks, I didn't know that.

**Pull request merged**